Week3/homework/7-step3.js

```diff
@@ -22,4 +26,6 @@ f2(y);
 console.log(y);
 
 // Add your explanation as a comment here
-
+//  In the case  console.log(x) is just simply asking to console log the value of  x which is 9  because the variable (global) x is assigned to number 9. 
+// In the case console.log(y); the variable y(global) is an abject with property x and its value number 9. There fore the result is   { x: 9 }. It is not related with the function calling. 
+// If the  console log (f2(y)):  the result will be different.
```

Comment:
I don't think you got the point of this exercise.
`x` is an atomic value i.e. `9`.
When you pass x into the `f1` function, it is passed as a value i.e. not a reference.  You then make changes to `val` within `f1` but this has no effect on `x` because `val` was a copy of `x`.  Printing `x` gives the original value of `9`.  If you had printed `val` it would be `10`.
`y` on the other hand is an object.and it is passed to `f2` as a reference.  So any changes made to items within the object, are reflected in `y`.  So, `f2` make changes to `val.x`, which is actually changing `y.x` as well.
When you print `y` after `f2` has run, you see the changes made in `f2` reflected in `y`.  And that is why `y.x` now shows `10`.